rustc_tools_util/src/lib.rs

```diff
@@ -100,19 +104,42 @@ impl std::fmt::Debug for VersionInfo {
 }
 
 #[must_use]
-pub fn get_commit_hash() -> Option<String> {
-    let output = std::process::Command::new("git")
-        .args(["rev-parse", "HEAD"])
+pub fn rerun_if_git_changes() -> Option<()> {
+    // Make sure we get rerun when the git commit changes.
+    // First, find the `HEAD` file. This should work even with worktrees.
+    let output = Command::new("git").args(["rev-parse", "--git-dir"]).output().ok()?;
+    let stdout = output.status.success().then_some(output.stdout)?;
+    let git_dir = PathBuf::from(String::from_utf8(stdout).ok()?.trim());
+    // Read the HEAD file to determine the name of the current ref.
+    let git_head_file = git_dir.join("HEAD");
+    let git_head_ref = String::from_utf8(std::fs::read(&git_head_file).ok()?).ok()?;
+    let git_head_ref = git_head_ref.strip_prefix("ref:")?.trim();
```

Comment:
This could be `git symbolic-ref -q HEAD`, we should also handle this failing separately since a detached head doesn't use a symbolic ref but we'd still want to print a `rerun-if-changed` for `HEAD`

Reply:
Okay, I am now using `symbolic-ref` instead of parsing the `HEAD` file.